### ISSUE TITLE
fix: ModelCache estabilidad — RAII guard, timer threads dangling, destructor race

### ISSUE THREAD

**Issue by SitoSt** [Jota-project/jota-transcriber#43]
## Rama: \`fix/modelcache-stability\`

Los tres bugs de \`ModelCache\` son interdependientes: deben arreglarse juntos.

---

### C1 — \`ModelCache.h\` Sin RAII guard para \`acquire()\` **(Crítico)**

\`ModelCache::instance().acquire()\` incrementa el refcount pero no existe ningún guard RAII. Si cualquier operación posterior lanza una excepción no capturada, el refcount queda incrementado para siempre → el modelo nunca se descarga de GPU.

\`InferenceLimiter\` tiene su propio \`Guard\` RAII; \`ModelCache\` no tiene equivalente.

**Fix:** Añadir \`ModelCache::Guard\` análogo al \`InferenceLimiter::Guard\`:
\`\`\`cpp
class Guard {
public:
    Guard(const std::string& path) { ctx_ = ModelCache::instance().acquire(path); }
    ~Guard() { ModelCache::instance().release(); }
    whisper_context* ctx() const { return ctx_; }
private:
    whisper_context* ctx_;
};
\`\`\`
Refactorizar \`HandleTranscribe\` para usar el guard.

---

### C4 — \`ModelCache.h:176\` Timer threads detached con \`this\` dangling **(Crítico)**

\`scheduleUnloadLocked\` hace \`std::thread(...).detach()\` capturando \`this\`. Si el singleton se destruye mientras el thread duerme, accede a memoria destruida.

**Fix:** Usar \`std::condition_variable\` con flag de cancelación. El destructor señaliza y espera a que el timer thread termine.

---

### I2 — \`ModelCache.h:144\` Destructor llama función "Locked" sin mutex **(Importante)**

\`cancelUnloadLocked()\` está diseñado para ser llamado con el mutex tomado. El destructor lo llama sin \`lock_guard\`.

**Fix:** Tomar el mutex en el destructor antes de llamar a \`cancelUnloadLocked()\`, o renombrar la función para reflejar que debe llamarse con o sin lock.

---

## Archivos a modificar
- \`src/whisper/ModelCache.h\`
- \`src/server/handlers/HandleTranscribe.cpp\` (usar el nuevo Guard)

## Tests a añadir
- Verificar que el refcount vuelve a 0 cuando se lanza excepción post-acquire
- Verificar que el destructor no hace data race con el timer thread
